OpenAI-DotNet/Chat/Function.cs

```diff
@@ -14,13 +27,18 @@ public Function(string name, string description = null, JsonObject parameters =
             if(name.Length > 64)
                 throw new ArgumentException("Function name cannot be longer than 64 characters.", nameof(name));
 
-            if (!char.IsLetter(name[0]) && name[0] != '_')
-                throw new ArgumentException("Function name must begin with a letter or underscore.", nameof(name));
+            // Not all possible "functions" must follow the following commented rules,
```

Comment:
Might be better if you just let the server validate it. 

If we can validate locally just use a regex expression